core/geometry/src/curve/internalContexts/CurveCurveIntersectXY.ts

```diff
@@ -1173,19 +1187,28 @@ export class CurveCurveIntersectXY extends RecurseToCurvesGeometryHandler {
         spiralFraction = newtonSearcher.getU();
         geometryBFraction = newtonSearcher.getV();
       }
-      if (this.acceptFraction(false, spiralFraction, false) && this.acceptFraction(false, geometryBFraction, false))
-        this.recordPointWithLocalFractions(spiralFraction, spiral, 0, 1, geometryBFraction, curvePrimitive, 0, 1, false);
+      if (this.acceptFraction(false, spiralFraction, false) && this.acceptFraction(false, geometryBFraction, false)) {
+        if (reversed)
+          this.recordPointWithLocalFractions(geometryBFraction, curvePrimitive, 0, 1, spiralFraction, spiral, 0, 1, false);
+        else
+          this.recordPointWithLocalFractions(spiralFraction, spiral, 0, 1, geometryBFraction, curvePrimitive, 0, 1, false);
```

Comment:
Can you take advantage of the reversed argument?
```suggestion
        this.recordPointWithLocalFractions(spiralFraction, spiral, 0, 1, geometryBFraction, curvePrimitive, 0, 1, reversed);
```